assets/app/scripts/services/applicationGenerator.js

```diff
@@ -0,0 +1,270 @@
+"use strict";
+
+angular.module("openshiftConsole")
+
+  .service("ApplicationGenerator", function(){
+    var osApiVersion = "v1beta1";
+    var k8sApiVersion = "v1beta3";
+    
+    var service = {};
+    var scope = service;
+    
+    scope._generateSecret_ = function(){
```

Comment:
The only reason they are accessible is to simply testing.  It allows us 
the ability to call them individually.

On 04/08/2015 03:15 PM, Jessica Forrester wrote:
> In assets/app/scripts/services/applicationGenerator.js
> <https://github.com/openshift/origin/pull/1221#discussion_r28003247>:
>
>> @@ -0,0 +1,270 @@
>> +"use strict";
>> +
>> +angular.module("openshiftConsole")
>> +
>> +  .service("ApplicationGenerator", function(){
>> +    var osApiVersion = "v1beta1";
>> +    var k8sApiVersion = "v1beta3";
>> +
>> +    var service = {};
>> +    var scope = service;
>> +
>> +    scope._generateSecret_ = function(){
>
> why are all these private methods on the scope object? unless the
> template needs to be able to access them (in which case it is strange
> that they are private functions) do |var _generateSecret = function| instead
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/1221/files#r28003247>.
>

-- 
Jeff Cantrill
Senior Software Engineer, Red Hat Engineering
Red Hat
Office: 703-748-4420 | 866-546-8970 ext. 8162420
jcantril@redhat.com
http://www.redhat.com